backend/kesaseteli/applications/views.py

```diff
@@ -163,3 +170,123 @@ def export_and_download_annual_applications(
             return self.render_error(_("Hakemuksia ei löytynyt."))
 
         return self.get_xlsx_response(queryset, columns)
+
+
+class YouthApplicationExcelExportView(APIView):
+    permission_classes = [AllowAny]  # Permissions are handled per function
+    http_method_names = ["get", "head", "options"]
+    serializer_class = YouthApplicationExcelExportSerializer
+
+    @staticmethod
+    def source_fields_and_output_names() -> List[Tuple[str, str]]:
+        with translation.override("fi"):
+            return [
+                ("application_date", str(_("Hakupvm"))),
+                ("id", str(_("Tunniste"))),
+                ("status", str(_("Hakemuksen tila"))),
+                ("application_year", str(_("Hakuvuosi"))),
+                ("summer_voucher_serial_number", str(_("Kesäsetelinro"))),
+                ("birth_year", str(_("Syntymävuosi"))),
+                ("birthdate", str(_("Syntymäpvm"))),
+                ("first_name", str(_("Etunimi"))),
+                ("last_name", str(_("Sukunimi"))),
+                ("vtj_last_name", str(_("VTJ-sukunimi"))),
+                ("school", str(_("Koulu"))),
+                ("is_unlisted_school", str(_("Listaamaton koulu?"))),
+                ("email", str(_("Sähköposti"))),
+                ("phone_number", str(_("Puhelinnro"))),
+                ("postcode", str(_("Postinro"))),
+                ("vtj_home_municipality", str(_("VTJ-kotikunta"))),
+                ("additional_info_user_reasons", str(_("Lisätietosyyt"))),
+                ("additional_info_description", str(_("Lisätiedot"))),
+                ("language", str(_("Kieli"))),
+                ("confirmation_date", str(_("Vahvistettu"))),
+                ("additional_info_providing_date", str(_("Lisätiedot annettu"))),
+                ("handling_date", str(_("Käsitelty"))),
+            ]
+
+    @staticmethod
+    def source_fields() -> Tuple[str, ...]:
+        return list(
+            zip(*YouthApplicationExcelExportView.source_fields_and_output_names())
+        )[0]
+
+    @staticmethod
+    def output_column_names() -> Tuple[str, ...]:
+        return list(
+            zip(*YouthApplicationExcelExportView.source_fields_and_output_names())
+        )[1]
```

Comment:
I didn't use dict because I thought it doesn't preserve the order but googling told that from [Python 3.7](https://docs.python.org/3/whatsnew/3.7.html) onwards
> the insertion-order preservation nature of dict objects has been declared to be an official part of the Python language spec.

so it could be used as we're using [Python 3.8](https://github.com/City-of-Helsinki/yjdh/blob/5f932b6c25499f9c8ef8aa7827e72b612a6c9573/backend/docker/kesaseteli.Dockerfile#L2) but I went with using OrderedDict to make it explicit that the preservation of the order of the insertion of the keys is needed i.e. the functionality will work wrong—the Excel output's columns will be in wrong order—without it.